src/rust/ide/view/graph-editor/src/component/node/input/area.rs

```diff
@@ -0,0 +1,661 @@
+                let port_network       = &port.network;
+                let frp                = &self.frp.output;
+
+                frp::extend! { port_network
+                    // === Aliases ===
+                    let mouse_over = port_shape.hover.events.mouse_over.clone_ref();
+                    let mouse_out  = port_shape.hover.events.mouse_out.clone_ref();
+                    let mouse_down = port_shape.hover.events.mouse_down.clone_ref();
+
+                    // === Press ===
+                    eval_ mouse_down ([crumbs,frp] frp.source.press.emit(&crumbs));
+
+                    // === Hover ===
+                    hover <- bool(&mouse_out,&mouse_over);
+                    self.source.port_hover <+ hover.map (f!([crumbs](t)(crumbs.clone_ref(),*t)));
+                    eval_ mouse_over ([crumbs,frp] frp.source.hover.emit(&Some(crumbs.clone())));
+                    eval_ mouse_out  (frp.source.hover.emit(&None));
+
+                    // === Pointer Style ===
+                    let port_shape_hover = port_shape.hover.shape.clone_ref();
+                    pointer_style_out   <- mouse_out.map(|_| default());
+                    pointer_style_over  <- map2(&mouse_over,&port.final_type,move |_,tp| {
+                        let color = tp.as_ref().map(|tp| type_coloring::compute(tp,&styles));
+                        let color = color.unwrap_or(missing_type_color);
+                        cursor::Style::new_highlight(&port_shape_hover,padded_size,Some(color))
+                    });
+                    pointer_style_hover <- any(pointer_style_over,pointer_style_out);
+                    pointer_styles      <- all[pointer_style_hover,self.model.label.pointer_style];
+                    pointer_style       <- pointer_styles.fold();
+                    self.frp.output.source.pointer_style <+ pointer_style;
+                }
+                port_shape.display_object().clone_ref()
+            };
+
+            if let Some(parent_frp) = &builder.parent_frp {
+                frp::extend! { port_network
+                    node.frp.set_active           <+ parent_frp.set_active;
+                    node.frp.set_hover            <+ parent_frp.set_hover;
+                    node.frp.set_parent_connected <+ parent_frp.set_parent_connected;
+                }
+            }
+            let new_parent_frp = Some(node.frp.output.clone_ref());
+            let new_shift = if !not_a_port { 0 } else { builder.shift + node.payload.local_index };
+            builder.nested(new_parent,new_parent_frp,is_parensed,new_shift)
+        });
+    }
+
+    /// Initializes FRP network for every port. Please note that the networks are connected
+    /// hierarchically (children get events from parents), so it is easier to init all networks
+    /// this way, rather than delegate it to every port.
+    fn init_port_frp_on_new_expression(&self, expression:&mut Expression) {
```

Comment:
If this is port FRP initialization, what is this network we build in build_port_shapes_on_new_expression?